src/coreclr/nativeaot/System.Private.CoreLib/src/System/Collections/Generic/Comparer.CoreRT.cs

```diff
@@ -48,8 +48,7 @@ internal sealed partial class EnumComparer<T> : Comparer<T> where T : struct, En
     {
```

Comment:
Do i need to do something special for `EnumComparer<T>` to generate `RuntimeConstructedGenericTypeInfo` and not `RuntimeBlockedTypeInfo`?

https://github.com/dotnet/runtimelab/blob/e5ab2d33f9d64e2ed407391c1b5b724f5c071f04/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/BlockedInternalsBlockingPolicy.cs#L124-L128

Gets blocked there since its internal and has nothing else that keeps it from being blocked, but quick search didn't show any special handling in CoreCLR

Reply:
I do not think you need to do anything special to make `EnumComparer` reflectable. It is supposed to be non-reflectable in NativeAOT.

Reply:
While that fine with me, it would mean that the test would still fail since it tries to compare the type
https://github.com/dotnet/runtimelab/blob/e5ab2d33f9d64e2ed407391c1b5b724f5c071f04/src/tests/JIT/opt/Devirtualization/Comparer_get_Default.cs#L221

Reply:
We can fixup the test. @MichalStrehovsky Do you have thoughts on the best way to do it?

Reply:
> We can fixup the test. @MichalStrehovsky Do you have thoughts on the best way to do it?

Nothing good comes to mind. The point of `EnumComparer` is that it doesn't allocate, so maybe we could test that calling `CompareTo` on it will not allocate?

Reply:
> point of EnumComparer is that it doesn't allocate, so maybe we could test that calling CompareTo on it will not allocate?

We typically cover this in performance tests, not in functional tests.

After thinking about it, I am leaning towards suppressing the tests when the type name is not available, e.g. only do this when ToString starts with `System.`.

Reply:
It can be done in separate PR.